uwblueprint/community-fridge-kw
# feat: food rescue volunteer sign up flow

## Brief description. What is this change? 

### [Food Rescue Volunteer Signup Flow](https://www.notion.so/uwblueprintexecs/Volunteer-Checkin-Shift-Signup-Flow-6147bb4ee0fc40af809aace5bf72355e)

This pull requests implements the screens for a volunteer to sign up for a food rescue or check-in. Additionally, it hooks up the components in the volunteer dashboard so that you can view your signed up shifts. 


## Implementation description. How did you make this change?
* Created the Food Rescues component that loads in all schedules that need volunteers
* Created the CheckIns component that populates a calendar component with checkins that need volunteers
   * Used useStep for navigation 
* Created new ConfirmShiftDetails component based on existing ConfirmDetails component and hooked it up to Volunteer Dashboard




https://user-images.githubusercontent.com/19617248/163681798-db287757-0b84-45ca-95fd-fda6db9c71c3.mp4


## Steps to test
1. This has latest changes merged in so you can create some checkins/food rescues
2. Verify you see them when you go to `/volunteer-shifts` 
3. Sign up for a shift
4. Confirm you can see shifts in your dashboard
5. Confirm you can delete your volunteer shifts and it appears again in the sign up flow


**Note: Cancelling checkin shifts seems to be not working because the checkin object has start date after end date:**


![image](https://user-images.githubusercontent.com/19617248/163681804-0b556b1a-7d21-4d1e-8f4a-6568cd345aa4.png)
